lib/c/criu.c

```diff
@@ -1657,3 +1657,68 @@ int criu_restore_child(void)
 {
 	return criu_local_restore_child(global_opts);
 }
+
+int criu_local_get_version(criu_opts *opts)
+{
+	int ret = -1;
+	CriuReq req	= CRIU_REQ__INIT;
+	CriuResp *resp	= NULL;
+
+	saved_errno = 0;
+
+	req.type	= CRIU_REQ_TYPE__VERSION;
+	req.opts	= opts->rpc;
+
+	ret = send_req_and_recv_resp(opts, &req, &resp);
+	if (ret)
+		goto exit;
+
+	if (resp->success) {
+		ret = resp->version->major_number * 10000;
+		ret += resp->version->minor_number * 100;
+		if (resp->version->has_sublevel)
+			ret += resp->version->sublevel;
+		if (resp->version->gitid) {
+			/* Taken from runc: a git release -> minor + 1 */
+			ret -= (ret % 100);
+			ret += 100;
+		}
+	} else {
+		ret = -EBADE;
```

Comment:
I don't like the idea to use both return value and errno for error codes. We have to choose only one of them.

Reply:
This code structure is just copied from all the other functions in the library. I have no problem changing it, just wanted to point out, that almost all functions in the library are structured like this. This behaviour is documented here https://github.com/checkpoint-restore/criu/blob/criu-dev/lib/c/criu.h#L120